<?xml version="1.0"?>

<!-- autoRallyTrackGazeboSim.launch
Use Gazebo to simulate a vehicle with Ackermann steering.
Arguments:
    namespace (string, default: auto_rally_platform)
        Vehicle namespace.
    world_name (string, default: worlds/empty.world)
        Gazebo world name.
    cmd_timeout (float, default: 0.5)
        Command timeout passed to the auto_rally_controller.py node.
    x (float, default: 0.0)
    y (float, default: 0.0)
    z (float, default: 0.1)
    roll (float, default: 0.0)
    pitch (float, default: 0.0)
    yaw (float, default: 0.0)
        Vehicle pose. x, y, and z are measured in meters. roll, pitch, and yaw
        are measured in radians.
Copyright (c) 2013 Wunderkammer Laboratory
Licensed under the Apache License, Version 2.0 (the "License");
you may not use this file except in compliance with the License.
You may obtain a copy of the License at
  http://www.apache.org/licenses/LICENSE-2.0
Unless required by applicable law or agreed to in writing, software
distributed under the License is distributed on an "AS IS" BASIS,
WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
See the License for the specific language governing permissions and
limitations under the License.
-->

<launch>

  <arg name="num_runs" default="1"/>

  <node name="autorally_random_pose" pkg="autorally_gazebo"
        type="autorally_random_pose.py" output="screen">
    <param name="num_runs" value="$(arg num_runs)"/>
    <param name="x_origin" value="0"/>
    <param name="y_origin" value="-8.5"/>
    <param name="map_path" value="$(env AR_MPPI_PARAMS_PATH)maps/marietta_costmap_09_08_2018.npz"/>
    <param name="image_left_topic" value="/left_camera/image_raw"/>
    <param name="image_right_topic" value="/right_camera/image_raw"/>
  </node>
</launch>
